internal/format/format.go

```diff
@@ -0,0 +1,104 @@
+package format
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+)
+
+type DataFormat string
+
+const (
+	FORMAT_LIST DataFormat = "list"
+	FORMAT_JSON DataFormat = "json"
+	FORMAT_YAML DataFormat = "yaml"
+)
+
+func (df DataFormat) String() string {
+	return string(df)
+}
+
+func (df *DataFormat) Set(v string) error {
+	switch DataFormat(v) {
+	case FORMAT_LIST, FORMAT_JSON, FORMAT_YAML:
+		*df = DataFormat(v)
+		return nil
+	default:
+		return fmt.Errorf("must be one of %v", []DataFormat{
+			FORMAT_LIST, FORMAT_JSON, FORMAT_YAML,
+		})
+	}
+}
+
+func (df DataFormat) Type() string {
+	return "DataFormat"
+}
+
+// MarshalData marshals arbitrary data into a byte slice formatted as outFormat.
+// If a marshalling error occurs or outFormat is unknown, an error is returned.
+//
+// Supported values are: json, list, yaml
+func Marshal(data interface{}, outFormat DataFormat) ([]byte, error) {
+	switch outFormat {
+	case FORMAT_JSON:
+		if bytes, err := json.MarshalIndent(data, "", "  "); err != nil {
+			return nil, fmt.Errorf("failed to marshal data into JSON: %w", err)
+		} else {
+			return bytes, nil
+		}
+	case FORMAT_YAML:
+		if bytes, err := yaml.Marshal(data); err != nil {
+			return nil, fmt.Errorf("failed to marshal data into YAML: %w", err)
+		} else {
+			return bytes, nil
+		}
+	case FORMAT_LIST:
+		return nil, fmt.Errorf("this data format cannot be marshaled")
+	default:
+		return nil, fmt.Errorf("unknown data format: %s", outFormat)
+	}
+}
+
+// UnmarshalData unmarshals a byte slice formatted as inFormat into an interface
+// v. If an unmarshalling error occurs or inFormat is unknown, an error is
+// returned.
+//
+// Supported values are: json, list, yaml
+func Unmarshal(data []byte, v interface{}, inFormat DataFormat) error {
```

Comment:
I noticed a discrepancy between the function comment and the function name, `UnmarshalData` versus `Unmarshal`.